src/main/java/fr/alasdiablo/janoeo/world/gen/feature/OresFeatures.java

```diff
@@ -838,136 +841,38 @@ public static void initNether() {
-                FREQUENCY_CONFIG.AMETHYST_ORE_BOTTOM.get(),
-                FREQUENCY_CONFIG.AMETHYST_ORE_TOP.get()
-        );
-
-        WorldGenerationHelper.ConfiguredFeatureHelper.registerOreFeature(
-                Objects.requireNonNull(
-                        OverworldOresBlocks.ZINC_ORE.getRegistryName()
-                ),
-                OreFeatureConfig.FillerBlockType.BASE_STONE_OVERWORLD,
-                OverworldOresBlocks.ZINC_ORE.getDefaultState(),
-                FREQUENCY_CONFIG.ZINC_ORE_SIZE.get(),
-                FREQUENCY_CONFIG.ZINC_ORE_COUNT.get(),
-                FREQUENCY_CONFIG.ZINC_ORE_BOTTOM.get(),
-                FREQUENCY_CONFIG.ZINC_ORE_TOP.get()
-        );
+        for (Map.Entry<String, Block> Ore : OverworldOresBlocks.Ores.entrySet()) {
+            String OreName = Ore.getKey();
+            Block OreBlock = Ore.getValue();
+
+            if (FREQUENCY_CONFIG.Frequency.containsKey("ALUMINIUM_ORE")) {
+                if (FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").containsKey("OVERWORLD.STONE")) {
+                    if (FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").containsKey("SIZE")) {
+                        WorldGenerationHelper.ConfiguredFeatureHelper.registerOreFeature(
+                                Objects.requireNonNull(
+                                        OreBlock.getRegistryName()
+                                ),
+                                OreFeatureConfig.FillerBlockType.BASE_STONE_OVERWORLD,
+                                OreBlock.getDefaultState(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("SIZE").get(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("COUNT").get(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("BOTTOM").get(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("TOP").get()
+                        );
+                    } else {
+                        WorldGenerationHelper.ConfiguredFeatureHelper.registerReplaceBlockFeature(
+                                Objects.requireNonNull(
+                                        OreBlock.getRegistryName()
+                                ),
+                                Blocks.STONE.getDefaultState(),
+                                OreBlock.getDefaultState(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("COUNT").get(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("BOTTOM").get(),
+                                FREQUENCY_CONFIG.Frequency.get("ALUMINIUM_ORE").get("OVERWORLD.STONE").get("TOP").get()
+                        );
+                    }
```

Comment:
The if condition is useless for Aluminium because we know he have a size.

Reply:
`ALUMINIUM_ORE` would be replaced with the value `oreName` which is used in `FREQUENCY_CONFIG.ORE_FREQUENCY` to get the FREQUENCY options for the current ore in the loop. However, I didn't setup the frequency configs for all of the ores at this time and this allowed all new ores to use ALIMININUM_ORE frequency settings.